src/test/java/org/datadog/jmxfetch/TestCommon.java

```diff
@@ -110,6 +110,16 @@ public void unregisterMBean() throws MBeanRegistrationException, InstanceNotFoun
         }
     }
 
+    /**
+     * Clear instances and their instance telemetry bean after execution of every test.
+     */
+    @After
+    public void clearInstances() {
+        if (app != null) {
+            app.clearInstances(getInstances());
```

Comment:
Lets make `protected void clearAllInstances()` instead of exposing `clearInstances` as `protected`. This can wrap up the call to `getInstances` as well.

Motivation here is to keep `clearInstances` as private and expose a more stable `protected` interface.